2312-ondo-contexts-added-on-step-through-not-removed

MarcusDenker opened the pull request:
better comment

MarcusDenker committed:
The  issue had already PR where this was tried to fix.... which was wrong.

- undo that change, put the  #terminateTo: in comments again in the case where the error was raised 
- add it to the case we see in the bug report: if no error hapens, we need to remove it!

fixes #2312

MarcusDenker committed:
better comment

myroslavarm reviewed (changes requested):
this makes sense although the second comment is a bit confusing:
"Error was *not* raised we need to it remove inserted above contexts then return signaler context"
was it supposed to be "we need to remove" without "it"?

MarcusDenker committed:
- played with debugging "self error" and the #terminateTo: is needed for the error case (else the on:do: stays)
-  step through on "self halt" needs another small fix to not bring the emergency evaluator
- fixed comment

MarcusDenker committed:
simplify comments

myroslavarm reviewed (approved):
good!

MarcusDenker commented:
Let's try it, we can always revert
